<launch>
  <arg name="uuv_name" />
  <arg name="start_time" default="-1"/>
  <arg name="radius" default="10"/>
  <arg name="center_x" default="0"/>
  <arg name="center_y" default="0"/>
  <arg name="center_z" default="-20"/>
  <arg name="n_points" default="50"/>
  <!-- Heading offset given in degrees -->
  <arg name="heading_offset" default="0"/>
  <arg name="duration" default="0"/>
  <arg name="max_forward_speed" default="0.3"/>
  <arg name="launch-prefix" default=""/>
  <arg name="respawn" default="false"/>

  <group ns="$(arg uuv_name)">
    <node pkg="eca_a9_gazebo"
      type="start_circular_trajectory_same_path.py"
      name="start_circular_trajectory_same_path"
      output="screen"
      launch-prefix="$(arg launch-prefix)"
      respawn="$(arg respawn)">
      <rosparam subst_value="true">
        start_time: $(arg start_time)
        radius: $(arg radius)
        center:
          - $(arg center_x)
          - $(arg center_y)
          - $(arg center_z)
        n_points: $(arg n_points)
        heading_offset: $(arg heading_offset)
        duration: $(arg duration)
        max_forward_speed: $(arg max_forward_speed)
      </rosparam>
    </node>
  </group>
</launch>
